### PR TITLE
Story 319: don't generate a new UUID if we were passed in a valid one

### PR DESCRIPTION
**Pull request by alecpm**
## Description
Changes `Experiment.make_uuid` to take an optional `app_id` parameter and if that parameter is a valid UUID, then it returns that instead of generating a new one. Existing code using `make_uuid` passes in the `app_id` when available.

## Motivation and Context
See [ScrumDo Story 319](https://app.scrumdo.com/projects/story_permalink/1536379)

## How Has This Been Tested?
Manual testing. Existing tests should cover all code changes.

### CODE REVIEW

File: dallinger/experiment.py [hunk 1]
```diff
@@ -471,8 +471,7 @@ def run(self, exp_config=None, app_id=None, bot=False, **kwargs):
         """
         import dallinger as dlgr
 
-        if app_id is None:
-            app_id = self.make_uuid()
+        app_id = self.make_uuid(str(app_id))
```
Comment (jessesnyder) [inline dallinger/experiment.py]: So if `app_id` is `None`, `str(app_id)` will return `"None"`, which will fail the `isinstance()` check inside `make_uuid()`, and then we'll generate one?

Comment (jessesnyder) [reply dallinger/experiment.py]: I wrote tests to satisfy my addled brain, so I committed and pushed. Maybe they'll fix the tiny coverage drop.

Comment (alecpm) [reply dallinger/experiment.py]: You know, I think moving the str() call into the make_uuid function makes more sense. I was trying to replicate some logic from command_line.py that was clearly put there for some edge case that I didn't want to dig into, but it does confuse the logic and result in an inefficiency.

Comment (alecpm) [reply dallinger/experiment.py]: You know, I think moving the str() call into the make_uuid function makes more sense. I was trying to replicate some logic from command_line.py that was clearly put there for some edge case that I didn't want to dig into, but it does confuse the logic and result in an inefficiency.

Thanks for the tests!!
